### PR TITLE
DNM: make zfs-mount service work with encryption

### PR DESCRIPTION
**Pull request by prometheanfire**

### CODE REVIEW

**Comment by prometheanfire** [comment]
the only reason this is DNM (do not merge) is because I can't test it

**Comment by behlendorf** [comment]
@prometheanfire is there still a need for these changes now that 7280d581973e923c6492b59803246a94347a47b8 was merged?

**Comment by prometheanfire** [comment]
I think it's still needed.  It is specifically useful for when you have a dataset you need but is not included in the rootfs dataset (maybe only /home is encrypted).

**Comment by behlendorf** [comment]
I see.  OK, let's get @kpande's thoughts on this.

File: etc/systemd/system/zfs-mount.service.in [hunk 1]
```diff
@@ -6,11 +6,16 @@ After=systemd-udev-settle.service
...
 After=systemd-remount-fs.service
 Before=local-fs.target
+StartLimitIntervalSec=60
+StartLimitBurst=5
 
 [Service]
 Type=oneshot
 RemainAfterExit=yes
+StandardInput=tty
+ExecStartPre=@bindir@/bash -c "if [ $(zpool list -H -o feature@encryption) == 'active' ]; then zfs load-key -a; fi
```
[the remark this answers is not in the document]
Comment (Fabian-Gruenbichler) [reply etc/systemd/system/zfs-mount.service.in]: would it not make more sense to put this into a separate service and just make `zfs load-key -a` a NOP that does not fail if encryption is not active/enabled/used/... (just like `zfs mount -a` and `zfs share -a` are NOPs if no datasets need to be mounted/shared)?

then `zfs-mount.service` can require `zfs-load-keys.service` and we don't need more shell hacks in systemd service files..

[the remark this answers is not in the document]
Comment (prometheanfire) [reply etc/systemd/system/zfs-mount.service.in]: probably

personally I'd like someone to take this over as I don't have a vested interest in it (and am not able to test this (which is the bigger issue)).

[the remark this answers is not in the document]
Comment (Fabian-Gruenbichler) [reply etc/systemd/system/zfs-mount.service.in]: I don't use (native) encryption (yet), but I can help with systemd related questions if needed.

I might have to install some dracut/CentOS system to investigate the dracut related systemd workaround with the BOOTFS environment variable, maybe I can tackle both things in one go - but no promises yet ;)

[the remark this answers is not in the document]
Comment (behlendorf) [reply etc/systemd/system/zfs-mount.service.in]: That does sounds better.  @Fabian-Gruenbichler if you do have time to look in to the best way to do this, that would be great.  In the meanwhile let's close this out until we decide exactly what should be done.

**Comment by prometheanfire** [comment]
sure, closing

**Pull request closed without merging**